Mathlib/Analysis/Cauchy.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Analysis.Normed.Field.Basic
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset NNReal
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
+  sup (range p.natDegree) (‖p.coeff ·‖₊) / ‖p.leadingCoeff‖₊
+
+lemma NNReal.geom_sum {x : ℝ≥0} (h : 1 < x) (n : ℕ) :
+    ∑ i ∈ Finset.range n, x ^ i = (x ^ n - 1) / (x - 1) := by
+  have : 0 ≠ x - 1 := fun nh ↦ (h.trans_le (tsub_eq_zero_iff_le.mp nh.symm)).false
+  field_simp
```

Comment:
```suggestion
  field_simp [(tsub_pos_of_lt h).ne']
```
though I'm not sure it's a good idea to have rigid code after `field_simp`, so alternatively
```suggestion
  apply eq_div_of_mul_eq (tsub_pos_of_lt h).ne'
```